### PR TITLE
support symfony 8

### PR DESCRIPTION
**Pull request by BordelD**
Type: feature
Issue: #903
Breaking change: no

### CODE REVIEW

**Review by AJenbo — DISMISSED** [review]

**Comment by AJenbo** [comment]
It looks like there are a few incompatibilities between 7 and 8 that has to be addressed before this will work properly.

**Comment by BordelD** [comment]
The 8 is dropping the XML loader: https://symfony.com/blog/new-in-symfony-7-4-deprecated-xml-configuration

Config needs to be translated to PHP or YAML first.

And the minimum PHP version required is 8.4.

**Comment by AJenbo** [comment]
Is this something you can handle?

**Comment by AJenbo** [comment]
That should do it, but it's dependant on https://github.com/pdepend/pdepend/pull/909 being merged first

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -60,5 +60,4 @@ jobs:
         run: composer test
 
       - name: Run PHPStan
-        if: matrix.php != '8.4'
```
Comment (kylekatarnls) [inline .github/workflows/tests.yml]: Unrelated but running PHPStan for each version is a waste, the result should be version-independant, some day we should give each static analyzer its own GH-Action workflow.

File: composer.json [hunk 2]
```diff
@@ -12,18 +12,19 @@
...
     },
     "require-dev": {
         "brianium/paratest": "^7.3",
         "easy-doc/easy-doc": "0.0.0 || ^1.2.3",
         "friendsofphp/php-cs-fixer": "^3.57",
         "gregwar/rst": "^1.0.6",
-        "phpstan/phpstan": "2.1.31",
-        "phpunit/phpunit": "^10.5.20,<10.5.32",
+        "phpstan/phpstan": "~2.1.32",
+        "phpunit/phpunit": "^10.5.20,<10.5.32|^11.0.0",
```
Comment (kylekatarnls) [inline composer.json]: Just for consistency with the rest of the file:
```suggestion
        "phpunit/phpunit": "^10.5.20,<10.5.32 || ^11.0.0",
```

**Review by kylekatarnls — DISMISSED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (AJenbo) [reply .github/workflows/tests.yml]: The analyzer is version aware so it does help with detecting edge cases that differs between PHP versions.

**Review by kylekatarnls — APPROVED** [review]
